### DISCUSSION TITLE
How to interop with Typescript

### DISCUSSION THREAD

**Discussion by Darkle** [eeue56/derw#15] [Q&A]
Hi, when using derw, how do you go about interoping with Typescript? Like for example if I want to use a 3rd party library written in typescript, is there some sort of FFI for me to put those Typescript types into my derw code so I can call out to the library?

Thanks.

**Comment by eeue56** [comment]
Hi there! So the Gitbook has some details on what Derw compiles to, typescript wise: https://docs.derw-lang.com/guides/creating-your-first-project

Generally for doing Typescript interop, you can just import things and use them as you generally would in TypeScript. However I recommend a pattern of having a typescript Kernel code that wraps the TS/JS interface in a way that Derw would like. There's some examples here: https://github.com/derw-lang/stdlib/tree/main/src (for example List). 

I'll get working on writing some docs on this so it's easier to see
